lms/djangoapps/certificates/tests/test_support_views.py

```diff
@@ -326,6 +334,38 @@ def test_regenerate_user_has_no_certificate(self):
         num_certs = GeneratedCertificate.eligible_certificates.filter(user=self.student).count()
         self.assertEqual(num_certs, 1)
 
+    def test_regenerate_user_with_invalidated_cert(self):
+        """ Verify that if certificate is marked as invalid than method will re-generate the
+        certificate and also de-activate the invalidated entry."""
```

Comment:
"Verify if the certificate is marked as invalid, if so, regenerate the certificate and de-activate the invalidated entry."